### DISCUSSION TITLE
How do I use the sqlite3 wrapper?

### DISCUSSION THREAD

**Discussion by algorithmiker** [duckdb/duckdb#6405] [Q&A]
Hey!
I am trying to FFI duckdb instead of sqlite into a niche programming language.
The problem is, it fails with `Invalid argument(s): Failed to lookup symbol 'sqlite3_open_v2':`
I suppose this is happening because the symbol names aren't consistent:

```
nm libsqlite3.so | grep sqlite3_open
0000000000112280 T sqlite3_open
00000000001122a0 T sqlite3_open16
0000000000112290 T sqlite3_open_v2
```

```
nm libsqlite3_api_wrapper.so | grep sqlite3_open

0000000000011350 T duckdb_shell_sqlite3_open
0000000000010f90 T duckdb_shell_sqlite3_open_v2
000000000000f297 t duckdb_shell_sqlite3_open_v2.cold
```

Why does this happen? 
I compiled with `make sqlite`.
What am I doing wrong?

**Comment by Mytherin** [comment]
To avoid conflicts between SQLite and DuckDB's SQLite wrapper in the shell the SQLite wrapper by default prefixes the methods with `duckdb_shell_`. This can be turned off by removing the `USE_DUCKDB_SHELL_WRAPPER` definition, [here](https://github.com/duckdb/duckdb/blob/b69925cac41a3bfe86a965a61e2fdbeb52dd9684/tools/sqlite3_api_wrapper/sqlite3/CMakeLists.txt#L4).

**Reply by algorithmiker** [reply]
Hey! Thanks for the answer. Sadly, compiling with this I get (while building):
```
[ 78%] Linking CXX executable test_sqlite3_api_wrapper
ld: libsqlite3_api_wrapper.so: undefined reference to `duckdb_shell_sqlite3_realloc64'
collect2: error: ld returned 1 exit status
make[3]: *** [tools/sqlite3_api_wrapper/CMakeFiles/test_sqlite3_api_wrapper.dir/build.make:126: tools/sqlite3_api_wrapper/test_sqlite3_api_wrapper] Error 1
make[2]: *** [CMakeFiles/Makefile2:5872: tools/sqlite3_api_wrapper/CMakeFiles/test_sqlite3_api_wrapper.dir/all] Error 2
```

**Reply by algorithmiker** [reply]
@Mytherin  I just commented out the line like this:
`#add_definitions(-DUSE_DUCKDB_SHELL_WRAPPER)`
Also, I did try deleting `build` and re-compiling again.
